contracts/DCNT721A.sol

```diff
@@ -78,13 +103,19 @@ contract DCNT721A is ERC721A, Initializable, Ownable, Splits {
         _metadataConfig.metadataRendererInit
       );
     } else {
+      _contractURI = _metadataConfig.contractURI;
       baseURI = _metadataConfig.metadataURI;
     }
   }
 
-  function mint(uint256 numberOfTokens) external payable {
-    uint256 mintIndex = totalSupply();
-    require(saleIsActive, "Sale must be active to mint");
+  function mint(uint256 numberOfTokens)
+    external
+    payable
+    verifyTokenGate(false)
+  {
+    uint256 mintIndex = _nextTokenId();
+    require(block.timestamp >= saleStart, "Sales are not active yet.");
+    require(!saleIsPaused, "Sale must be active to mint");
```

Comment:
can we abstract these requires to a `onlyActive` modifier just like you have in Crescendo?

Also, been seeing Zora using `revert` + `errror` instead of `require` not sure if there's a benefit but pointing this out as something that's been on my mind. 💭

Reply:
sure, single line used only twice... if once 👎, if twice 🤷‍♂️, if thrice 👍

fwiw, in crescendo, the salesAreActive modifier checks both saleStart as well as saleIsPaused, and is used by both buy and sell

in the case of 721, saleIsPaused is used by mint and mintPresale, but they separately check sale start date and presale window, so not a dead match for the crescendo modifier

primary advantage of revert is being able to nest it within more complex logic, vs a require statement

for custom errors, biggest improvement there is taking the strings out of the bytecode, custom errors are just method signatures.

i'm all for custom errors, but feel they should be adopted in a holistic way, or just moving forward on new contracts, as opposed to 1 error at a time.